libtrap/src/ifc_file.c

```diff
@@ -559,37 +597,44 @@ int file_send(void *priv, const void *data, uint32_t size, int timeout)
    /* Writes data_length bytes to the file */
    written = fwrite(data, 1, size, config->fd);
    if (written != size) {
-      VERBOSE(CL_ERROR, "OUTPUT FILE IFC: unable to write to file: %s", config->filename);
-      return trap_errorf(config->ctx, TRAP_E_IO_ERROR, "OUTPUT FILE IFC: unable to write");
+      return trap_errorf(config->ctx, TRAP_E_IO_ERROR, "FILE OUTPUT IFC[%u]: unable to write to file: %s", config->ifc_idx, config->filename);
    }
 
    if (config->file_change_time != 0) {
       time_t current_time = time(NULL);
-      if (difftime(current_time, config->starting_time) / 60 >= config->file_change_time) {
-         char *new_filename = create_filename_from_time(priv);
-         if (!new_filename) {
-            VERBOSE(CL_ERROR, "OUTPUT FILE IFC[%d]: memory allocation failed.", config->ifc_idx);
-            return trap_error(config->ctx, TRAP_E_MEMORY);
+
+      /* Check whether new file should be created */
+      if (difftime(current_time, config->create_time) / 60 >= config->file_change_time) {
+
+         /* Create new filename from the current timestamp */
+         int status = create_next_filename(priv);
+         if (status != TRAP_E_OK) {
+            return trap_errorf(config->ctx, status, "FILE OUTPUT IFC[%u]: Error during output file creation.", config->ifc_idx);
          }
 
-         if (open_next_file(priv, new_filename) < 0) {
-            VERBOSE(CL_ERROR, "OUTPUT FILE IFC[%d]: opening new file failed.", config->ifc_idx);
-            return trap_errorf(config->ctx, TRAP_E_BADPARAMS, "unable to open file");
+         /* Open newly created file */
+         status = open_next_file(priv, config->filename);
+         if (status != TRAP_E_OK) {
+            return trap_errorf(config->ctx, status, "FILE OUTPUT IFC[%u]: Error during output file opening.", config->ifc_idx);
          }
       }
+
+      config->file_index = 0;
    }
 
    if (config->file_change_size != 0 && (uint64_t)ftell(config->fd) >= (uint64_t)(1024 * 1024 * (uint64_t)config->file_change_size)) {
-      char *new_filename = create_next_filename(priv);
-      if (!new_filename) {
-         VERBOSE(CL_ERROR, "OUTPUT FILE IFC[%d]: memory allocation failed.", config->ifc_idx);
-         return trap_error(config->ctx, TRAP_E_MEMORY);
+
+      /* Create new filename from the current timestamp */
+      int status = create_next_filename(priv);
+      if (status != TRAP_E_OK) {
+         return trap_errorf(config->ctx, status, "FILE OUTPUT IFC[%u]: Error during output file creation.", config->ifc_idx);
       }
 
-         if (open_next_file(priv, new_filename) < 0) {
-            VERBOSE(CL_ERROR, "OUTPUT FILE IFC[%d]: opening new file failed.", config->ifc_idx);
-            return trap_errorf(config->ctx, TRAP_E_BADPARAMS, "unable to open file");
-         }
+      /* Open newly created file */
+      status = open_next_file(priv, config->filename);
```

Comment:
Again - open_next_file() outside open_next_file_wrapper()

Reply:
explained in different comment